ui/page/proposal/proposal_vote_modal.go

```diff
@@ -1,191 +1,409 @@
-		value, err := strconv.Atoi(i.input.Editor.Text())
-		if err != nil {
-			return
-		}
-		value++
-		i.input.Editor.SetText(fmt.Sprintf("%d", value))
+func (vm *voteModal) eligibleVotes() int {
+	vm.detailsMu.Lock()
+	voteDetails := vm.voteDetails
+	vm.detailsMu.Unlock()
+
+	if voteDetails == nil {
+		return 0
 	}
 
-	if i.decrement.Button.Clicked() {
-		value, err := strconv.Atoi(i.input.Editor.Text())
-		if err != nil {
-			return
-		}
-		value--
-		if value < 0 {
-			return
-		}
-		i.input.Editor.SetText(fmt.Sprintf("%d", value))
+	return len(voteDetails.EligibleTickets)
+}
+
+func (vm *voteModal) remainingVotes() int {
+	vm.detailsMu.Lock()
+	voteDetails := vm.voteDetails
+	vm.detailsMu.Unlock()
+
+	if voteDetails == nil {
+		return 0
 	}
 
-	if i.max.Button.Clicked() {
-		i.input.Editor.SetText("5")
+	return len(voteDetails.EligibleTickets) - (vm.yesVote.voteCount() + vm.noVote.voteCount())
+}
+
+func (vm *voteModal) sendVotes() {
+	vm.detailsMu.Lock()
+	tickets := vm.voteDetails.EligibleTickets
+	vm.detailsMu.Unlock()
+
+	votes := make([]*dcrlibwallet.ProposalVote, 0)
+	addVotes := func(bit string, count int) {
+		for i := 0; i < count; i++ {
+
+			// get and pop
+			var eligibleTicket *dcrlibwallet.EligibleTicket
+			eligibleTicket, tickets = tickets[0], tickets[1:]
+
+			vote := &dcrlibwallet.ProposalVote{
+				Ticket: eligibleTicket,
+				Bit:    bit,
+			}
+
+			votes = append(votes, vote)
+		}
 	}
+
+	addVotes(dcrlibwallet.VoteBitYes, vm.yesVote.voteCount())
+	addVotes(dcrlibwallet.VoteBitNo, vm.noVote.voteCount())
+
+	modal.NewPasswordModal(vm.Load).
+		Title("Confirm to vote").
+		NegativeButton("Cancel", func() {
+			vm.isVoting = false
+		}).
+		PositiveButton("Confirm", func(password string, pm *modal.PasswordModal) bool {
+			go func() {
+				err := vm.WL.MultiWallet.Politeia.CastVotes(vm.walletSelector.selectedWallet.ID, votes, vm.proposal.Token, password)
+				if err != nil {
+					pm.SetError(err.Error())
+					pm.SetLoading(false)
+					return
+				}
+				pm.Dismiss()
+				vm.CreateToast("Vote sent successfully, refreshing proposals!", true)
+				go vm.WL.MultiWallet.Politeia.Sync()
+				vm.Dismiss()
+			}()
+
+			return false
+		}).Show()
 }
 
-func (cm *voteModal) Handle() {
-	if cm.btnNegative.Button.Clicked() {
-		cm.Dismiss()
+func (vm *voteModal) Handle() {
+	for vm.cancelBtn.Button.Clicked() {
+		if vm.isVoting {
+			continue
+		}
+		vm.Dismiss()
 	}
 
-	cm.yesVote.handleVoteCountButtons()
-	cm.noVote.handleVoteCountButtons()
+	vm.handleVoteCountButtons(vm.yesVote)
+	vm.handleVoteCountButtons(vm.noVote)
+
+	totalVotes := vm.yesVote.voteCount() + vm.noVote.voteCount()
+	validToVote := totalVotes > 0 && totalVotes <= vm.eligibleVotes()
+	vm.voteBtn.SetEnabled(validToVote)
+
+	for vm.voteBtn.Clicked() {
+		if vm.isVoting {
+			break
+		}
+
+		if !validToVote {
+			break
+		}
+
+		vm.isVoting = true
+		vm.sendVotes()
+	}
 }
 
-func (cm *voteModal) Layout(gtx layout.Context) D {
+// - Layout
```

Comment:
Was this left here on purpose? does not look like a complete comment to me.. if not, we can remove this.

Reply:
This is used to divide the go file into the frontend and backend sections of a page.